pkg/serverless/apikey/api_key.go

```diff
@@ -30,6 +31,11 @@ const encryptionContextKey = "LambdaFunctionName"
 // functionNameEnvVar is the environment variable that stores the function name.
 const functionNameEnvVar = "AWS_LAMBDA_FUNCTION_NAME"
 
+// one of those env variable must be set
+const apiKeyEnvVar = "DD_API_KEY"
+const apiKeySecretManagerEnvVar = "DD_API_KEY_SECRET_ARN"
+const apiKeyKmsEncryptedEnvVar = "DD_API_KEY_KMS_ENCRYPTED"
```

Comment:
I think we're still assuming/setting DD_KMS_API_KEY in our IAC integrations. Does this PR break that? Or is this being copied/migrated in the config file some way?

[serverless plugin](https://github.com/DataDog/serverless-plugin-datadog/blob/a724ffc9f212b4b9f264f5d3b5c97c8455a7c921/src/env.ts#L121)
[CFN Macro](https://github.com/DataDog/datadog-cloudformation-macro/blob/48e7758961c22756f7ca22c41dea87c4b9d4ae3c/serverless/src/env.ts#L85)
[CDK Construct](https://github.com/DataDog/datadog-cdk-constructs/blob/f9ef48a4a846ef3939288e04371b3a5a39ce0e15/v2/src/transport.ts#L15)

Reply:
no we would need to change those or re-enable support for this env in the extension

Reply:
I think in that case we should re-enable support for this `DD_KMS_API_KEY` here, no?

otherwise we'd have to coordinate breaking changes across 3 other repositories.

Reply:
yeah I think you're right, let's create a ticket in those 3 repos to track that? (to make sure we won't support this dual env forever?

Reply:
Do we know if it ever worked here? I see `DD_KMS_API_KEY` in every layer, deployment tool, the [datadog forwarder](https://github.com/search?q=repo%3ADataDog%2Fdatadog-serverless-functions%20DD_KMS_API_KEY&type=code), and our own docs.

I don't see the variable `DD_API_KEY_KMS_ENCRYPTED` referenced anywhere, it will be much easier to drop it here.

In fact I only see the reference in the [original PR](https://github.com/DataDog/datadog-agent/pull/12155). Maybe I'm missing something?

Reply:
I think it was added to allow other env var to be decrypted by KMS (not only this one)

Reply:
cc @IvanTopolcic

Reply:
I can't find any documentation for this [at all](https://github.com/search?q=repo%3ADataDog%2Fdocumentation%20DD_API_KEY_KMS_ENCRYPTED&type=code) so to me this seems like a misunderstanding and this env var was never released.

Either way I think we should deprecate this, and favor `DD_KMS_API_KEY` instead.

Reply:
I see it for other configurations but not this: https://github.com/DataDog/documentation/blob/028596ee37a3d4028bd01f0bd6dabd36e038710e/content/en/serverless/aws_lambda/configuration.md?plain=1#L830

Reply:
This is used in dual shipping (tab KMS) https://docs.datadoghq.com/serverless/aws_lambda/configuration/?tab=awskms#send-telemetry-to-multiple-datadog-organizations

Reply:
so if I have an array of `DD_API_KEY_KMS_ENCRYPTED: [key1, key2]` this functionality works? I don't see that in the docs you've linked but I believe the functionality works.

Regardless, we can support both, but I don't think we should indicate that `DD_KMS_API_KEY` is deprecated in code or elsewhere, if anything we should deprecate `DD_API_KEY_KMS_ENCRYPTED` as we don't document that nor honor it anywhere else.